### PR TITLE
Add check for Phase correlation transform

### PR DESCRIPTION
**Pull request by ruqayya**
Likewise other intermediate transform, Phase cross-correlation transform should only be applied if it improves the DICE overlap.

### CODE REVIEW

File: tiatoolbox/tools/registration/wsi_registration.py [hunk 1]
```diff
@@ -1107,6 +1107,14 @@ def register(
             fixed_tissue_img, moving_tissue_img
         )
         translation_offset = np.array([[1, 0, shift[1]], [0, 1, shift[0]], [0, 0, 1]])
+        transform_tissue_mask = apply_affine_transformation(
+            fixed_tissue_mask, moving_tissue_mask, translation_offset
+        )
+
+        # Use the estimated phase cross correlation transform only if it improves DICE overlap
```
Comment (shaneahmed) [inline tiatoolbox/tools/registration/wsi_registration.py:1114]:
```suggestion
        # Use the estimated phase cross correlation transform 
        # only if it improves DICE overlap
```

**Review by shaneahmed — COMMENTED** [review]
I think as there is an if statement you may need to add a test. 
You can create a new function that would help to add a quick test.

**Comment by ruqayya** [comment]
> I think as there is an if statement you may need to add a test. You can create a new function that would help to add a quick test.

'If statement' is already covered by the existing test functions such as test_register_output_with_initializer. The phase cross correlation transform for the images that we are using for testing results in low dice value as compared to previous dice value and hence "if statement" is executed on calling register function. Do we still need to have a separate function?

**Review by shaneahmed — APPROVED** [review]
Thanks @ruqayya

**Pull request merged**